### PR TITLE
PUBDEV-3527 Check that Java version is 1.6-1.8 and exit otherwise

### PR DESCRIPTION
**Pull request by mstensmo**
This change is [Reviewable](https://reviewable.io/reviews/h2oai/h2o-3/310)

### CODE REVIEW

File: h2o-core/src/main/java/water/H2O.java [hunk 1]
```diff
@@ -1728,6 +1728,12 @@ public static void gc() {
   // --------------------------------------------------------------------------
   public static void main( String[] args ) {
 
+    String javaVersion = System.getProperty("java.version");
+    if (!(javaVersion.startsWith("1.6") || javaVersion.startsWith("1.7") || javaVersion.startsWith("1.8"))) {
```
Comment (mmalohlava) [inline h2o-core/src/main/java/water/H2O.java]: It is better to check for `null` as well - some obscure JVMs do not need to setup the property.

File: h2o-core/src/main/java/water/H2O.java [hunk 2]
```diff
@@ -1725,9 +1723,30 @@ public static void gc() {
     }
   }
 
+  /**
+   * Check if the Java version is not supported
+   * @return true if not supported
+   */
+  public static boolean checkUnsupportedJava() {
+    String version = System.getProperty("java.version");
+    if (version != null && !(version.startsWith("1.6") || version.startsWith("1.7") || version.startsWith("1.8"))) {
```
Comment (mdymczyk) [inline h2o-core/src/main/java/water/H2O.java]: This is ok now with the null check but for the future you can also use `org.apache.commons.lang.SystemUtils.IS_JAVA_1_6` etc.

File: h2o-core/src/main/java/water/H2O.java (hunk 1 above)
Comment (mstensmo) [inline h2o-core/src/main/java/water/H2O.java]: done

File: h2o-core/src/main/java/water/H2O.java (hunk 2 above)
Comment (mmalohlava) [inline h2o-core/src/main/java/water/H2O.java]: Agree, i would stay in plain java here - the library can always fail on load...(using static variable reading from vendor specific mx beans)

**Pull request merged**